webpack/components/AnsibleHostDetail/components/AnsibleHostInventory/index.js

```diff
@@ -0,0 +1,20 @@
+import React, { useMemo } from 'react';
+import { foremanUrl } from 'foremanReact/common/helpers';
+import { useAPI } from 'foremanReact/common/hooks/API/APIHooks';
+import PropTypes from 'prop-types';
+import AnsibleHostInventory from './AnsibleHostInventory';
+
+const WrappedAnsibleHostInventory = ({ hostId }) => {
+  const params = useMemo(() => ({ params: { host_ids: [hostId] } }), [hostId]);
+
+  const url = foremanUrl('/ansible/api/ansible_inventories/hosts');
+  const { response: inventory } = useAPI('get', url, params);
+
+  return <AnsibleHostInventory inventoryData={inventory} />;
+};
+
+WrappedAnsibleHostInventory.propTypes = {
+  hostId: PropTypes.number.isRequired,
```

Comment:
the host id can be undefined in the first rendering, can you remove the  `isRequired` and add a default?

Reply:
@amirfefer , if id is not available, then the component should wait for it to be present. Supplying a fake hostId value (which should be never used to call API!) seems like an awkward thing to do and it tells me we need to reconsider how we compose things together.

Reply:
`undefined` is not a fake id, if the host id is undefined no ajax call will be triggered.